api/src/main/java/org/apache/iceberg/metrics/MetricsReporter.java

```diff
@@ -18,10 +18,23 @@
  */
 package org.apache.iceberg.metrics;
 
+import java.util.Map;
+
 /** This interface defines the basic API for reporting metrics for operations to a Table. */
 @FunctionalInterface
 public interface MetricsReporter {
 
+  /**
+   * Initialize a MetricsReporter with catalog properties.
+   *
+   * <p>A custom MetricsReporter implementation must have a no-arg constructor. After initialize the
+   * MetricsReporter without any arguments, this method should be called to complete MetricsReporter
+   * initialization with catalog properties.
+   *
+   * @param properties catalog properties
```

Comment:
These aren't necessarily catalog properties. Just "properties" is fine.

Reply:
done